site/docs/components/_content-template/index.mdx

```diff
@@ -0,0 +1,208 @@
+                     _contains_ List)
+        - similarTo: This component is similar to the other component (e.g. a
+                     Switch _similarTo_ Checkbox)
+      */
+      relationship: "contains",
+    },
+    {
+      name: "Another Component Name",
+      relationship: "similarTo",
+    },
+    // ...
+  ],
+};
+
+# Component Name
+
+<!--
+  Example screenshot or live component demo here of the default/most common
+  usage.
+
+  For screenshots use Markdown:
+
+  ![Text alternative for screenshot](./path/to/screenshot.jpg)
+
+  For live demos in Storybook use:
+
+  <Canvas>
+    <Story id="..." />
+  </Canvas>
+-->
+
+<!--
+  Introductory text that briefly explains the purpose of the component and
+  anything else of note.
+-->
+
+## Usage
+
+### When to use Component Name
+
+<!-- Bulleted list of scenarios in which this component can be used  -->
+
+- ...
+- ...
+
+### When not to use Component Name
+
+<!--
+  Bulleted list of scenarios where this component is not appropriate. Where
+  possible, include a link to an alternative component that should be used in
+  it's place
+-->
+
+- ...
+- ...
+
+### References
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  If there are industry standards for this component, list them in this
+  section. Include links out to useful articles.
+-->
+
+- [Relevant industry standard](https://example-standard.org/)
+- [Useful article](https://some-site.com/article/bla)
+- ...
+
+## Configurations
+
+<!--
+  A visual list of possible configurations of the component.
+
+  The purpose is to convey the main features and/or usage options of the
+  component (e.g. the Data Grid's striped rows might be a "configuration" to
+  showcase here). The configurations listed here should aim to be comprehensive,
+  but not necessarily exhaustive (some complex components may have vast numbers
+  of possible permutations, so listing out each one would be impossible and
+  overwhelming to readers).
+
+  Each confguration should have its own section following the structure outlined
+  below:
+-->
+
+### Configuration title (e.g. Striped rows)
+
+<!--
+  A brief description of what this configuration provides to users and when it
+  should be used.
+-->
+
+<!--
+  A live example of the component configuration, with access to the correspoding
+  source code.
+
+  In Storybook this is done using something like:
+
+  <Canvas>
+    <Story id="..." />
+  </Canvas>
+-->
+
+## Content
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  Add any component-specific content guidance here, use visual examples where
+  appropriate.
+
+  For example, a Button component might include guidance on recommended terms to
+  use for the label text, recommended maximum character counts, etc.
+-->
+
+## Accessibility
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  Describe any considerations the users may need to take into account when using
+  the component.
```

Comment:
Maybe we could add some guidance about what is worth documenting, assuming we don't want every minute detail documented. These things come to mind as of value:

- Accessibility decisions that deviate from industry standard. (e.g., we've introduced an arrow key navigation, where usually is only tab).
- Accessibility requirements that drive design (e.g., how multi-column forms should wrap).
- Best practices regarding accessible content (e.g., error message announcements must not rely on descriptions that require the user to visually or audibly perceive content ).
- Best practices regarding implementation (e.g., to make sure this component is accessible be sure to provide a relevant alt text).

Reply:
Thanks. I think I can add your suggestions almost verbatim!

Reply:
Does make me wonder about where accessibility should live though... @shinytoyrobots mentioned it shouldn't have its own section/tab. I agree with that perspective (since it touches on different aspects) _but_ then i find myself thinking about this as a template. Interspersing accessibility content (or call outs) seems like an opportunity for inconsistency to creep in, whilst a single section or tab, would be straightforward. Thoughts?